model/DataObject.php

```diff
@@ -423,6 +423,20 @@ private static function cache_composite_fields($class) {
 	}
 
 	/**
+	 * Creates a class instance by the "singleton" design pattern.
+	 * It will always return the same instance for this class,
+	 * which can be used for performance reasons and as a simple
+	 * way to access instance methods which don't rely on instance
+	 * data (e.g. the custom SilverStripe static handling).
+	 *
+	 * @return static The singleton instance
+	 */
+	public static function singleton() {
```

Comment:
My mistake, that's what I get for trusting my memory.

I did commit this to master, but this was back when master WAS 3.2 :D